### ISSUE TITLE
Enrollment fails if email address has more than one client_id associated in user_clients table

### ISSUE THREAD

**Issue by phirework** [mozilla/voice-web#2359]

**Comment by mbransn** [comment]
@rileyjshaw @phirework I'd like to level this up to P0 as I'm uncomfortable with the potential of how many duplicate accounts may be signing up from the Mozilla side for the challenge.

Is it possible to scrub all duplicate accounts? What are the contingencies and possible consequences of doing so?

**Comment by phirework** [comment]
Should be fixed by #2378, pending code review. 

The TLDR is that in Nov 2018, we switched to using email addresses as the somewhat-unique-identifier for users, and introduced a new field called `has_login` to keep track of it. If you had an account on the site prior to that, it is 1) possible that you had more than one entry in the database because of how we were previously keeping track of unique users, and 2) the new field would be set to empty for those accounts. If you then logged in again after November 2018, instead of trying to update one of the existing entries, it would just create a new entry that has `has_login = true` and use that as your canonical account. There are no duplicate emails in the `has_login = true` set, so I've just added that as a check to enrolling users. 

In terms of the scrubbing the duplicate accounts, my expectation is that there would be knock-on effects through all other user actions - we would have to update/merge the client_id in every single table where it's referenced (which is most of them), including votes and clips, and then also make the same updates to all the S3 clips, which are not only stored in a folder associated with the client_id but the client_id is actually hardcoded into the name of the file. Theoretically possible, but a huge lift for not much gain. It would be more important to make sure our internal utilities are ironclad so that in the future we're can using our own tested functions instead of raw queries, which would reduce the chance of these sorts of errors being introduced. 

cc @rileyjshaw

**Comment by rileyjshaw** [comment]
#2378 is now on staging!

**Comment by mbransn** [comment]
Confirmed fixed for my LDAP account on stage! Nice work y’all.
